master/internal/user/service.go

```diff
@@ -107,6 +107,30 @@ func (s *Service) ProcessAuthentication(next echo.HandlerFunc) echo.HandlerFunc
 	}
 }
 
+// ProcessProxyAuthentication is a middleware processing function that attempts
+// to authenticate incoming HTTP requests coming through proxies.
+func (s *Service) ProcessProxyAuthentication(c echo.Context) (done bool, err error) {
+	token, err := s.extractToken(c)
+	if err != nil {
+		return true, c.Redirect(
+			http.StatusSeeOther,
+			fmt.Sprintf("/det/login?redirect=%s", c.Request().URL.Path),
+		)
+	}
+
+	switch user, _, err := s.db.UserByToken(token, s.extConfig); err {
```

Comment:
At least in isolation, I think the failure cases here (except maybe the default?) should also redirect instead of throwing a hard error. Though maybe that doesn't match up with the behavior elsewhere.